runtime/hydradx/src/evm/aave_trade_executor.rs

```diff
@@ -195,6 +196,9 @@ where
 		let decoded = decode(&param_types, reserves_data.as_ref())
 			.map_err(|_| ExecutorError::Error("Failed to decode reserves list".into()))?;
 
+		// ensure sufficient length
+		ensure!(decoded.len() >= 1, ExecutorError::Error("Empty reserve list".into()));
```

Comment:
```suggestion
		ensure!(decoded.len() == param_types.len(), ExecutorError::Error("Empty reserve list".into()));
```

Reply:
i considered it. not sure if there can be more or not. so feel like having >= does not hurt.